client/landing/gutenboarding/onboarding-block/signup-form/index.tsx

```diff
@@ -32,14 +32,25 @@ const SignupForm = () => {
 	const isFetchingNewUser = useSelect( select => select( USER_STORE ).isFetchingNewUser() );
 	const newUser = useSelect( select => select( USER_STORE ).getNewUser() );
 	const newUserError = useSelect( select => select( USER_STORE ).getNewUserError() );
-	const { shouldCreate } = useSelect( select => select( ONBOARD_STORE ) ).getState();
+	const { shouldCreate, siteTitle, siteVertical } = useSelect( select =>
+		select( ONBOARD_STORE )
+	).getState();
 
 	const history = useHistory();
 
 	const handleSignUp = ( event: React.FormEvent< HTMLFormElement > ) => {
 		event.preventDefault();
 
-		createAccount( { email: emailVal, is_passwordless: true, signup_flow_name: 'gutenboarding' } );
+		const username_hint = siteTitle || siteVertical?.label;
```

Comment:
Do you think we should do a character count hint here?

In the backend, once invalid characters are stripped, it falls back to an email-based username if there are fewer than 4 alphanumeric characters in the name.

Reply:
I'd kinda prefer to keep the logic on the backend so one day we decide the limit should change to 5 characters for example.
Is the idea of doing the character count here to save bytes if we knew the hint wasn't going to be used?

Reply:
> I'd kinda prefer to keep the logic on the backend so one day we decide the limit should change to 5 characters for example.

That makes sense. 

In that case we'll need need to make sure we don't fall back to the email address for Gutenboarding in the backend, but instead use our random username generator.

> Is the idea of doing the character count here to save bytes if we knew the hint wasn't going to be used?

No, just to skive out of the backend changes mentioned above 😁

Reply:
Forget what I just said. 

I actually did some testing and see that we do fall back to the random generator. 

Sorry for the confusion!

Reply:
Aaah ok! I misunderstood, and now understand your misunderstanding, so I think we're all good now. Understand?

Reply:
As long as you understand, I'll just ✅ away 😆